lib/bamboo/email.ex

```diff
@@ -1,8 +1,8 @@
 defmodule Bamboo.Email do
   defstruct from: nil,
-      to: nil,
-      cc: nil,
-      bcc: nil,
+      to: [],
+      cc: [],
+      bcc: [],
```

Comment:
After thinking about this more, I think this should be how it's handled. If it starts as nil and you set an list then you probably meant to do that. I can see it going either way though

Here are a couple scenarios:

1) You are sending emails to people subcribed to a post. In your tests in dev you set up a post and have people subscribed and everything works just fine. Then in production you have posts with no subscribers. Would you want it to raise? I think I would expect it to just not send an email

2) You want to send to just one user, in which case you would not have a list. The only error I can imagine would be accidentally passing in nil. e.g. `Emails.successfully_published(article.author)` where the articles author is nil. In which case I *would* expect it to raise. 

So I think I'm going to go ahead with keep the default addresses as nil, and raising only if all of them are nil, and nooping if the the recipients are an empty array